### PR TITLE
Updated Editor to 2017.3.0p2

### PR DESCRIPTION
**Pull request by StephenHodgson**
- Fixes: #1552 Removed previous Editor compatibility.
- Fixes: #1548 Updated the Editor to 2017.3.0p2
- Fixes: #1398 Fixes debugging on HoloLens
- Shader Upgrade NOTE: upgraded instancing buffer 'Props' to new syntax.
- Remove stabilization plane logic
- Add app setup wizard support for enabling the shared depth buffer.
- Remove duplicate SDK verification in build window utilities.

TODO
---

- [x] Write a check to see if shared depth buffer is enabled when the editor starts and UWP build target is specified.

Known Issues
---

[`WWW` and `UnityWebRequests` have false positive errors for UWP apps built against x86](https://issuetracker.unity3d.com/issues/wsa-isnetworkerror-always-return-true-when-running-an-uwp-app-on-86x-architecture).

### CODE REVIEW

File: Assets/HoloToolkit/UX/Shaders/3DTextShader.shader [hunk 1]
```diff
@@ -1,6 +1,8 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
+// Upgrade NOTE: upgraded instancing buffer 'Props' to new syntax.
```
Comment (timGerken) [inline Assets/HoloToolkit/UX/Shaders/3DTextShader.shader]: Do we need to keep these upgrade notes?

Comment (StephenHodgson) [reply Assets/HoloToolkit/UX/Shaders/3DTextShader.shader]: I thought they might be useful, but i can removed if needed.

Comment (timGerken) [reply Assets/HoloToolkit/UX/Shaders/3DTextShader.shader]: Maybe add the comment to the PR description.
Every time Unity goes and changes their shader syntax, these notes get littered throughout the affected shaders. Over time, they lose their cachet.

**Comment by DDReaper** [comment]
Are these shader changes, breaking changes?  So what happens if you use an older version of 2017?
P.S.
I was going to ask whether we were updating :S

**Comment by timGerken** [comment]
@DDReaper Good Question. It looks like these changes were introduced in [2017.3.0b4](https://unity3d.com/unity/beta/unity2017.3.0b4)...

**Comment by StephenHodgson** [comment]
Indeed.  That is a good point.

**Comment by StephenHodgson** [comment]
Yup. It's a breaking change.
image

**Comment by StephenHodgson** [comment]
If that's the case we should just drop all the support for prev versions of the editor as well.

**Comment by StephenHodgson** [comment]
Going to reopen this when I finish the work needed for 1552

**Comment by StephenHodgson** [comment]
image

**Comment by DDReaper** [comment]
We really should remove the ProjectVersion file and just keep the version tag on the github page stating supported versions.
Seen this done in other Unity projects to avoid constantly having to update the file.

**Comment by StephenHodgson** [comment]
I agree, but some people made a good point, that in the unitypackage, there's nothing that really states what the current MRTK version is, and when people inherit project's they're sometimes left wondering.

**Comment by DDReaper** [comment]
I created a "Package.Json" file in the root of some of my packages to resolve that, using standard notation.   It was suggested to me by the community
e.g.
https://bitbucket.org/UnityUIExtensions/unity-ui-extensions/src/d3671c3711b8e4078fdb271a91fc14e84c363611/package.json?at=master&fileviewer=file-view-default

**Comment by StephenHodgson** [comment]
@NeerajW was this the option you were talking about during the shiproom meeting?
image

**Comment by StephenHodgson** [comment]
@NeerajW I looked at figuring out how to update the Depth Buffer Sharing in the editor, but it seems like Unity did not provide an easy way to do so when setting the Virtual Reality SDK.

It also seems that this is not enabled by default.

**Comment by keveleigh** [comment]
@DDReaper That Package.Json looks way nicer than the [txt file](https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/Dev_Working_Branch/Assets/HoloToolkit/MRTKVersion.txt) I added 😆

I don't think I agree with removing ProjectVersion.txt though, since Unity can use that to update project settings, APIs, shaders, etc between versions, like we saw when moving to 2017.3. I'm not sure I trust it to guess which version the project is currently on before attempting to update the project.

**Comment by DDReaper** [comment]
Other projects I've worked with @keveleigh removed it because Unity automatically does that upgrade regardless.  When it does a library rebuild, it automatically runs the script updater to check no matter what is recorded in the projectversion file.

The only thing that file does is to inform the Project Wizard what it was last built against :D 

Plus, in the asset, we don't use it anyway

**Comment by keveleigh** [comment]
Sweet, I think this is just missing the readme updates where the version is referenced. Otherwise, looks good!

**Comment by StephenHodgson** [comment]
Still waiting to address the Stabilization Plane.

**Comment by StephenHodgson** [comment]
Blocked until stabilization plane is removed and app setup wizard supports enabling the shared depth buffer

**Comment by StephenHodgson** [comment]
Something else to consider is the verifying installed SDK reflection methods have been changed as well too.

**Comment by StephenHodgson** [comment]
@NeerajW , @keveleigh This PR is now ready for review.

Tested out the HoloLens version with the shared depth buffer. Couldn't really tell the difference. 

Maybe others can test this out as well.

**Comment by StephenHodgson** [comment]
After a bit of testing I ran into issues with the camera.
```
Failed to get rendering parameters for main camera - view transforms may get out of sync! Error code: '0x80004002'
 
(Filename: C:\buildslave\unity\build\Runtime/VR/HoloLens/VRDeviceHoloLens.cpp Line: 554)
```

**Pull request closed without merging**